### PR TITLE
ft(updateProfile):update-user-profile-#bL7qBcWY

### PR DESCRIPTION
**Pull request by Mugema123**
**What does this PR do?**
This PR update Driver or Operator Profile
**Description of Task to be completed?**
creating a controller for a updating driver or operator
Testing the controller
creating a validation for controller

**How should this be manually tested?**
-clone this repo
-run npm i
-Create a sendgrid account and it key in the .env
-run npm run dev
-Get all users
*localhost:3020/api/v1/auth/allusers
-Login as a Driver or Operator
*localhost:3020/api/v1/auth/login
-Update any field
*localhost:3020/api/v1/auth/updateprofile/:id

**What are the relevant pivotal tracker stories?**
#bL7qBcWY
w
w1
w2

### CODE REVIEW

**Comment by Ally4** [comment]
That's good Ronald!. I would like to ask you to check on the deployment!.

**Review by Ntwali-Josue — APPROVED** [review]

**Comment by Ntwali-Josue** [comment]
resolve these conflicts @Mugema123

**Comment by Ally4** [comment]
work on the conflict

**Review by Ally4 — APPROVED** [review]

**Review by TuyizeeAnastase — APPROVED** [review]

**Comment by blaise82** [comment]
Make sure that it doesn't return admins and password while trying to fetch drivers and operators

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ services:
 before_script:
   - psql -c 'CREATE DATABASE test;' -U postgres
   - psql -c 'CREATE DATABASE users;' -U postgres
+  - psql -c 'CREATE DATABASE andela;' -U postgres
+  - psql -c 'CREATE DATABASE test_db;' -U postgres
```
Comment (blaise82) [inline .travis.yml]: make sure that only one database creation 
is added to your .Travis.yml file

**Comment by blaise82** [comment]
make sure that a user can't update his email or role

File: src/controllers/userController.js [hunk 2]
```diff
@@ -95,5 +95,52 @@ class register {
...
+    if (updatedField) {
+      const updatedUser = await Users.findOne({ where: { email },
+        attributes:{
+          exclude:['password']
+        }
+      });
+
+      return res.status(200).json({ 
+        status:'successfull updated',
+        user: updatedUser })
```
Comment (blaise82) [inline src/controllers/userController.js]: you first need to check if the user exists 
then if he does then update the user 
or return 404

**Review by blaise82 — CHANGES_REQUESTED** [review]

**Review by blaise82 — APPROVED** [review]

**Pull request merged**